### DISCUSSION TITLE
Consider moving to C++ Modules (when modules are ready)

### DISCUSSION THREAD

**Discussion by LouChiSoft** [RavEngine/RavEngine#12] [Ideas]
C++ modules are slowly reaching a point where they are usable. Once that is the case it would be nice to move to a module based engine that does away with `#include` where possible

For context, all the work I have been doing using the engine have been included in modules themselves. So to have the entire codebase be module based would be great

**Comment by Ravbug** [comment]
C++ Modules are definitely something we want to support. However, they are not yet in a state where they can support all of this project's needs. Once they have matured enough, we'll definitely look into switching to them.

**Reply by LouChiSoft** [reply]
That's about what I expect. But I am glad you have already considered it.
